ush/ioda/bufr2ioda/bufr2ioda_subpfl_glider_profiles.py

```diff
@@ -0,0 +1,318 @@
+
+    # =====================================
+    # Create IODA ObsSpace
+    # Write IODA output
+    # =====================================
+
+    # Create the dimensions
+    dims = {'Location': np.arange(0, lat.shape[0])}
+
+    iodafile = f"{cycle_type}.t{hh}z.{data_type}_profiles.{data_format}.nc"
+    OUTPUT_PATH = os.path.join(ioda_dir, iodafile)
+    logger.debug(f" ... ... Create OUTPUT file: {OUTPUT_PATH}")
+
+    path, fname = os.path.split(OUTPUT_PATH)
+    if path and not os.path.exists(path):
+        os.makedirs(path)
+
+    obsspace = ioda_ospace.ObsSpace(OUTPUT_PATH, mode='w', dim_dict=dims)
+
+    # Create Global attributes
+    logger.debug(f" ... ... Create global attributes")
+    obsspace.write_attr('Converter', converter)
+    obsspace.write_attr('source', source)
+    obsspace.write_attr('sourceFiles', bufrfile)
+    obsspace.write_attr('dataProviderOrigin', data_provider)
+    obsspace.write_attr('description', data_description)
+    obsspace.write_attr('datetimeRange', [str(dateTime.min()), str(dateTime.max())])
+    obsspace.write_attr('platformLongDescription', platform_description)
+
+    # Create IODA variables
+    logger.debug(f" ... ... Create variables: name, type, units, and attributes")
+
+    # Datetime
+    obsspace.create_var('MetaData/dateTime', dtype=dateTime.dtype, fillval=dateTime.fill_value) \
+        .write_attr('units', 'seconds since 1970-01-01T00:00:00Z') \
+        .write_attr('long_name', 'Datetime') \
+        .write_data(dateTime)
+
+    # rcptDatetime
+    obsspace.create_var('MetaData/rcptdateTime', dtype=dateTime.dtype, fillval=dateTime.fill_value) \
+        .write_attr('units', 'seconds since 1970-01-01T00:00:00Z') \
+        .write_attr('long_name', 'receipt Datetime') \
+        .write_data(rcptdateTime)
+
+    # Longitude
+    obsspace.create_var('MetaData/longitude', dtype=lon.dtype, fillval=lon.fill_value) \
+        .write_attr('units', 'degrees_east') \
+        .write_attr('valid_range', np.array([-180, 180], dtype=np.float32)) \
+        .write_attr('long_name', 'Longitude') \
+        .write_data(lon)
+
+    # Latitude
+    obsspace.create_var('MetaData/latitude', dtype=lat.dtype, fillval=lat.fill_value) \
+        .write_attr('units', 'degrees_north') \
+        .write_attr('valid_range', np.array([-90, 90], dtype=np.float32)) \
+        .write_attr('long_name', 'Latitude') \
+        .write_data(lat)
+
+    # Station Identification
+    obsspace.create_var('MetaData/stationID', dtype=stationID.dtype, fillval=stationID.fill_value) \
+        .write_attr('long_name', 'Station Identification') \
+        .write_data(stationID)
+
+    # Depth
+    obsspace.create_var('MetaData/depth', dtype=pressure.dtype, fillval=pressure.fill_value) \
+        .write_attr('units', 'm') \
+        .write_attr('long_name', 'Water depth') \
+        .write_data(pressure)
+
+    # Sequence Number
+    obsspace.create_var('MetaData/sequenceNumber', dtype=PreQC.dtype, fillval=PreQC.fill_value) \
```

Comment:
Confirming that you want this to have the same dtype as PreQC and not sequenceNumber specific?

Reply:
yes.